scheduler/scheduler-examples/src/main/java/org/ow2/proactive/scheduler/examples/FTPConnector.java

```diff
@@ -0,0 +1,390 @@
+        File localFile = new File(ftpLocalFilePath);
+        // if the ftp remote file path does not exist then it is created.
+        if (!checkDirectoryExists(ftpClient, ftpRemoteFilePath)) {
+            ftpClient.makeDirectory(ftpRemoteFilePath);
+            getOut().println(ftpRemoteFilePath + " does not exist. Therefore, it is created.");
+        } else {
+            getOut().println(ftpRemoteFilePath + " exists.");
+        }
+        // If it is a single file:
+        if (localFile.isFile()) {
+            getOut().println("A single FILE to upload");
+            ftpClient.changeWorkingDirectory(ftpRemoteFilePath);
+            boolean uploaded = uploadSingleFile(ftpClient, ftpLocalFilePath, localFile.getName());
+            if (uploaded) {
+                getOut().println("UPLOADED a file to: " + Paths.get(ftpRemoteFilePath, localFile.getName()));
+            } else {
+                getOut().println("COULD NOT upload the file: " + ftpLocalFilePath);
+            }
+        }
+        // If it is a folder, upload all its contents recursively
+        else if (localFile.isDirectory()) {
+            getOut().println("A DIRECTORY to upload");
+            ftpClient.changeWorkingDirectory(ftpRemoteFilePath);
+            uploadDirectory(ftpClient, "", ftpLocalFilePath, "");
+        } else {
+            throw new IllegalArgumentException(localFile + " not found. Please, enter a valid path.");
+        }
+        getOut().println("END Export file(s) to FTP.");
+    }
+
+    private void uploadDirectory(FTPClient ftpClient, String remoteDirPath, String localParentDir,
+            String remoteParentDir) throws IOException {
+
+        getOut().println("LISTING directory: " + localParentDir);
+
+        File localDir = new File(localParentDir);
+        File[] subFiles = localDir.listFiles();
+        if (subFiles != null && subFiles.length > 0) {
+            for (File item : subFiles) {
+                uploadSubFilesDirectory(ftpClient, remoteDirPath, localParentDir, remoteParentDir, item);
+            }
+        }
+    }
+
+    private void uploadSubFilesDirectory(FTPClient ftpClient, String remoteDirPath, String localParentDir,
+            String remoteParentDir, File item) throws IOException {
+        String remoteFilePath = Paths.get(remoteDirPath, remoteParentDir, item.getName()).toString();
+        if (remoteParentDir.equals("")) {
+            remoteFilePath = Paths.get(remoteDirPath, item.getName()).toString();
+        }
+
+        if (item.isFile()) {
+            // upload the file
+            String localFilePath = item.getAbsolutePath();
+            getOut().println("About to upload the file: " + localFilePath);
+            boolean uploaded = uploadSingleFile(ftpClient, localFilePath, remoteFilePath);
+            if (uploaded) {
+                getOut().println("UPLOADED a file to: " + remoteFilePath);
+            } else {
+                getOut().println("COULD NOT upload the file: " + localFilePath);
```

Comment:
same remark as above, at least this behavior should be configurable.

It could be a number of retry for example, to retry several times, and then after all attempts have fail, then fail the task.

Reply:
done